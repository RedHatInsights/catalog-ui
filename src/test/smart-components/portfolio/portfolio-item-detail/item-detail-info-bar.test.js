import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import ItemDetailInfoBar from '../../../../smart-components/portfolio/portfolio-item-detail/item-detail-info-bar';

describe('<ItemDetailInfoBar />', () => {
  let initialProps;
  beforeAll(() => {
    Date.now = jest
      .spyOn(Date, 'now')
      .mockImplementation(() =>
        new Date(
          'Fri Mar 22 2019 08:36:57 GMT+0100 (Central European Standard Time)'
        ).getTime()
      );
  });
  beforeEach(() => {
    initialProps = {
      product: {
        distributor: 'foo',
        service_offering_source_ref: '111',
        updated_at:
          'Fri Mar 22 2019 08:36:57 GMT+0100 (Central European Standard Time)',
        created_at:
          'Fri Mar 22 2019 08:36:57 GMT+0100 (Central European Standard Time)'
      },
      source: {
        id: '111',
        name: 'bar'
      },
      portfolio: {
        display_name: 'baz',
        name: 'quux'
      }
    };
  });

  it('should render correctly', () => {
    const wrapper = mount(<ItemDetailInfoBar {...initialProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly with fallback values without vendor', () => {
    initialProps = {
      product: {
        created_at:
          'Fri Mar 22 2019 08:36:57 GMT+0100 (Central European Standard Time)',
        service_offering_source_ref: '111'
      },
      source: {
        id: '111',
        name: 'bar'
      },
      portfolio: {
        name: 'quux'
      }
    };
    const wrapper = mount(<ItemDetailInfoBar {...initialProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly with vendor', () => {
    const wrapper = mount(
      <ItemDetailInfoBar {...initialProps} distributor="Foo distributor" />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
