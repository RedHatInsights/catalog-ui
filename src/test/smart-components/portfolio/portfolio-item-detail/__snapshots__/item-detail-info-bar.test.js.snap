// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemDetailInfoBar /> should render correctly 1`] = `
<ItemDetailInfoBar
  portfolio={
    Object {
      "display_name": "baz",
      "name": "quux",
    }
  }
  product={
    Object {
      "created_at": "Fri Mar 22 2019 08:36:57 GMT+0100 (Central European Standard Time)",
      "distributor": "foo",
      "service_offering_source_ref": "111",
      "updated_at": "Fri Mar 22 2019 08:36:57 GMT+0100 (Central European Standard Time)",
    }
  }
  source={
    Object {
      "id": "111",
      "name": "bar",
    }
  }
>
  <Component
    className="pf-u-mb-md"
  >
    <div
      className="pf-c-content pf-u-mb-md"
    >
      <Component
        className="font-14"
      >
        <p
          className="font-14"
          data-pf-content={true}
        >
          Platform
        </p>
      </Component>
      <Component
        className="overflow-wrap"
        component="p"
        id="source-name"
      >
        <p
          className="overflow-wrap"
          data-pf-content={true}
          id="source-name"
        >
          bar
        </p>
      </Component>
      <Component
        className="font-14"
      >
        <p
          className="font-14"
          data-pf-content={true}
        >
          Portfolio
        </p>
      </Component>
      <Component
        className="overflow-wrap"
        component="p"
        id="portfolio-name"
      >
        <p
          className="overflow-wrap"
          data-pf-content={true}
          id="portfolio-name"
        >
          quux
        </p>
      </Component>
      <span
        id="distributor"
      >
        <Component
          className="font-14"
        >
          <p
            className="font-14"
            data-pf-content={true}
          >
            Vendor
          </p>
        </Component>
        <Component
          className="overflow-wrap"
          component="p"
        >
          <p
            className="overflow-wrap"
            data-pf-content={true}
          >
            foo
          </p>
        </Component>
      </span>
      <Component
        className="font-14"
      >
        <p
          className="font-14"
          data-pf-content={true}
        >
          Created
        </p>
      </Component>
      <Component
        component="p"
        id="created_at"
      >
        <p
          className=""
          data-pf-content={true}
          id="created_at"
        >
          <DateFormat
            date="Fri Mar 22 2019 08:36:57 GMT+0100 (Central European Standard Time)"
            variant="relative"
          >
            <Tooltip
              appendTo={[Function]}
              aria="describedby"
              boundary="window"
              className=""
              content={
                <div>
                  
                  22 Mar 2019 07:36 UTC
                </div>
              }
              distance={15}
              enableFlip={true}
              entryDelay={500}
              exitDelay={500}
              flipBehavior={
                Array [
                  "top",
                  "right",
                  "bottom",
                  "left",
                  "top",
                  "right",
                  "bottom",
                ]
              }
              id=""
              isAppLauncher={false}
              isContentLeftAligned={false}
              isVisible={false}
              maxWidth="18.75rem"
              position="top"
              tippyProps={Object {}}
              trigger="mouseenter focus"
              zIndex={9999}
            >
              <PopoverBase
                appendTo={[Function]}
                aria="describedby"
                arrow={true}
                boundary="window"
                content={
                  <div
                    className=""
                    id=""
                    role="tooltip"
                  >
                    <Unknown
                      isLeftAligned={false}
                    >
                      <div>
                        
                        22 Mar 2019 07:36 UTC
                      </div>
                    </Unknown>
                  </div>
                }
                delay={
                  Array [
                    500,
                    500,
                  ]
                }
                distance={15}
                flip={true}
                flipBehavior={
                  Array [
                    "top",
                    "right",
                    "bottom",
                    "left",
                    "top",
                    "right",
                    "bottom",
                  ]
                }
                isVisible={false}
                lazy={true}
                maxWidth="18.75rem"
                onCreate={[Function]}
                placement="top"
                popperOptions={
                  Object {
                    "modifiers": Object {
                      "hide": Object {
                        "enabled": true,
                      },
                      "preventOverflow": Object {
                        "enabled": true,
                      },
                    },
                  }
                }
                theme="pf-tooltip"
                trigger="mouseenter focus"
                zIndex={9999}
              >
                <span>
                  Just now
                </span>
                <Portal
                  containerInfo={
                    <div>
                      <div
                        class=""
                        id=""
                        role="tooltip"
                      >
                        <div
                          class="pf-c-tooltip__content"
                        >
                          <div>
                            
                            22 Mar 2019 07:36 UTC
                          </div>
                        </div>
                      </div>
                    </div>
                  }
                >
                  <div
                    className=""
                    id=""
                    role="tooltip"
                  >
                    <Component
                      isLeftAligned={false}
                    >
                      <div
                        className="pf-c-tooltip__content"
                      >
                        <div>
                          22 Mar 2019 07:36 UTC
                        </div>
                      </div>
                    </Component>
                  </div>
                </Portal>
              </PopoverBase>
            </Tooltip>
          </DateFormat>
        </p>
      </Component>
    </div>
  </Component>
</ItemDetailInfoBar>
`;

exports[`<ItemDetailInfoBar /> should render correctly with fallback values without vendor 1`] = `
<ItemDetailInfoBar
  portfolio={
    Object {
      "name": "quux",
    }
  }
  product={
    Object {
      "created_at": "Fri Mar 22 2019 08:36:57 GMT+0100 (Central European Standard Time)",
      "service_offering_source_ref": "111",
    }
  }
  source={
    Object {
      "id": "111",
      "name": "bar",
    }
  }
>
  <Component
    className="pf-u-mb-md"
  >
    <div
      className="pf-c-content pf-u-mb-md"
    >
      <Component
        className="font-14"
      >
        <p
          className="font-14"
          data-pf-content={true}
        >
          Platform
        </p>
      </Component>
      <Component
        className="overflow-wrap"
        component="p"
        id="source-name"
      >
        <p
          className="overflow-wrap"
          data-pf-content={true}
          id="source-name"
        >
          bar
        </p>
      </Component>
      <Component
        className="font-14"
      >
        <p
          className="font-14"
          data-pf-content={true}
        >
          Portfolio
        </p>
      </Component>
      <Component
        className="overflow-wrap"
        component="p"
        id="portfolio-name"
      >
        <p
          className="overflow-wrap"
          data-pf-content={true}
          id="portfolio-name"
        >
          quux
        </p>
      </Component>
      <Component
        className="font-14"
      >
        <p
          className="font-14"
          data-pf-content={true}
        >
          Created
        </p>
      </Component>
      <Component
        component="p"
        id="created_at"
      >
        <p
          className=""
          data-pf-content={true}
          id="created_at"
        >
          <DateFormat
            date="Fri Mar 22 2019 08:36:57 GMT+0100 (Central European Standard Time)"
            variant="relative"
          >
            <Tooltip
              appendTo={[Function]}
              aria="describedby"
              boundary="window"
              className=""
              content={
                <div>
                  
                  22 Mar 2019 07:36 UTC
                </div>
              }
              distance={15}
              enableFlip={true}
              entryDelay={500}
              exitDelay={500}
              flipBehavior={
                Array [
                  "top",
                  "right",
                  "bottom",
                  "left",
                  "top",
                  "right",
                  "bottom",
                ]
              }
              id=""
              isAppLauncher={false}
              isContentLeftAligned={false}
              isVisible={false}
              maxWidth="18.75rem"
              position="top"
              tippyProps={Object {}}
              trigger="mouseenter focus"
              zIndex={9999}
            >
              <PopoverBase
                appendTo={[Function]}
                aria="describedby"
                arrow={true}
                boundary="window"
                content={
                  <div
                    className=""
                    id=""
                    role="tooltip"
                  >
                    <Unknown
                      isLeftAligned={false}
                    >
                      <div>
                        
                        22 Mar 2019 07:36 UTC
                      </div>
                    </Unknown>
                  </div>
                }
                delay={
                  Array [
                    500,
                    500,
                  ]
                }
                distance={15}
                flip={true}
                flipBehavior={
                  Array [
                    "top",
                    "right",
                    "bottom",
                    "left",
                    "top",
                    "right",
                    "bottom",
                  ]
                }
                isVisible={false}
                lazy={true}
                maxWidth="18.75rem"
                onCreate={[Function]}
                placement="top"
                popperOptions={
                  Object {
                    "modifiers": Object {
                      "hide": Object {
                        "enabled": true,
                      },
                      "preventOverflow": Object {
                        "enabled": true,
                      },
                    },
                  }
                }
                theme="pf-tooltip"
                trigger="mouseenter focus"
                zIndex={9999}
              >
                <span>
                  Just now
                </span>
                <Portal
                  containerInfo={
                    <div>
                      <div
                        class=""
                        id=""
                        role="tooltip"
                      >
                        <div
                          class="pf-c-tooltip__content"
                        >
                          <div>
                            
                            22 Mar 2019 07:36 UTC
                          </div>
                        </div>
                      </div>
                    </div>
                  }
                >
                  <div
                    className=""
                    id=""
                    role="tooltip"
                  >
                    <Component
                      isLeftAligned={false}
                    >
                      <div
                        className="pf-c-tooltip__content"
                      >
                        <div>
                          22 Mar 2019 07:36 UTC
                        </div>
                      </div>
                    </Component>
                  </div>
                </Portal>
              </PopoverBase>
            </Tooltip>
          </DateFormat>
        </p>
      </Component>
    </div>
  </Component>
</ItemDetailInfoBar>
`;

exports[`<ItemDetailInfoBar /> should render correctly with vendor 1`] = `
<ItemDetailInfoBar
  distributor="Foo distributor"
  portfolio={
    Object {
      "display_name": "baz",
      "name": "quux",
    }
  }
  product={
    Object {
      "created_at": "Fri Mar 22 2019 08:36:57 GMT+0100 (Central European Standard Time)",
      "distributor": "foo",
      "service_offering_source_ref": "111",
      "updated_at": "Fri Mar 22 2019 08:36:57 GMT+0100 (Central European Standard Time)",
    }
  }
  source={
    Object {
      "id": "111",
      "name": "bar",
    }
  }
>
  <Component
    className="pf-u-mb-md"
  >
    <div
      className="pf-c-content pf-u-mb-md"
    >
      <Component
        className="font-14"
      >
        <p
          className="font-14"
          data-pf-content={true}
        >
          Platform
        </p>
      </Component>
      <Component
        className="overflow-wrap"
        component="p"
        id="source-name"
      >
        <p
          className="overflow-wrap"
          data-pf-content={true}
          id="source-name"
        >
          bar
        </p>
      </Component>
      <Component
        className="font-14"
      >
        <p
          className="font-14"
          data-pf-content={true}
        >
          Portfolio
        </p>
      </Component>
      <Component
        className="overflow-wrap"
        component="p"
        id="portfolio-name"
      >
        <p
          className="overflow-wrap"
          data-pf-content={true}
          id="portfolio-name"
        >
          quux
        </p>
      </Component>
      <span
        id="distributor"
      >
        <Component
          className="font-14"
        >
          <p
            className="font-14"
            data-pf-content={true}
          >
            Vendor
          </p>
        </Component>
        <Component
          className="overflow-wrap"
          component="p"
        >
          <p
            className="overflow-wrap"
            data-pf-content={true}
          >
            foo
          </p>
        </Component>
      </span>
      <Component
        className="font-14"
      >
        <p
          className="font-14"
          data-pf-content={true}
        >
          Created
        </p>
      </Component>
      <Component
        component="p"
        id="created_at"
      >
        <p
          className=""
          data-pf-content={true}
          id="created_at"
        >
          <DateFormat
            date="Fri Mar 22 2019 08:36:57 GMT+0100 (Central European Standard Time)"
            variant="relative"
          >
            <Tooltip
              appendTo={[Function]}
              aria="describedby"
              boundary="window"
              className=""
              content={
                <div>
                  
                  22 Mar 2019 07:36 UTC
                </div>
              }
              distance={15}
              enableFlip={true}
              entryDelay={500}
              exitDelay={500}
              flipBehavior={
                Array [
                  "top",
                  "right",
                  "bottom",
                  "left",
                  "top",
                  "right",
                  "bottom",
                ]
              }
              id=""
              isAppLauncher={false}
              isContentLeftAligned={false}
              isVisible={false}
              maxWidth="18.75rem"
              position="top"
              tippyProps={Object {}}
              trigger="mouseenter focus"
              zIndex={9999}
            >
              <PopoverBase
                appendTo={[Function]}
                aria="describedby"
                arrow={true}
                boundary="window"
                content={
                  <div
                    className=""
                    id=""
                    role="tooltip"
                  >
                    <Unknown
                      isLeftAligned={false}
                    >
                      <div>
                        
                        22 Mar 2019 07:36 UTC
                      </div>
                    </Unknown>
                  </div>
                }
                delay={
                  Array [
                    500,
                    500,
                  ]
                }
                distance={15}
                flip={true}
                flipBehavior={
                  Array [
                    "top",
                    "right",
                    "bottom",
                    "left",
                    "top",
                    "right",
                    "bottom",
                  ]
                }
                isVisible={false}
                lazy={true}
                maxWidth="18.75rem"
                onCreate={[Function]}
                placement="top"
                popperOptions={
                  Object {
                    "modifiers": Object {
                      "hide": Object {
                        "enabled": true,
                      },
                      "preventOverflow": Object {
                        "enabled": true,
                      },
                    },
                  }
                }
                theme="pf-tooltip"
                trigger="mouseenter focus"
                zIndex={9999}
              >
                <span>
                  Just now
                </span>
                <Portal
                  containerInfo={
                    <div>
                      <div
                        class=""
                        id=""
                        role="tooltip"
                      >
                        <div
                          class="pf-c-tooltip__content"
                        >
                          <div>
                            
                            22 Mar 2019 07:36 UTC
                          </div>
                        </div>
                      </div>
                    </div>
                  }
                >
                  <div
                    className=""
                    id=""
                    role="tooltip"
                  >
                    <Component
                      isLeftAligned={false}
                    >
                      <div
                        className="pf-c-tooltip__content"
                      >
                        <div>
                          22 Mar 2019 07:36 UTC
                        </div>
                      </div>
                    </Component>
                  </div>
                </Portal>
              </PopoverBase>
            </Tooltip>
          </DateFormat>
        </p>
      </Component>
    </div>
  </Component>
</ItemDetailInfoBar>
`;
